### PR TITLE
feat: new permissions system

### PR DESCRIPTION
**Pull request by marianogoldman**

### CODE REVIEW

File: src/adapters/worlds-manager.ts [hunk 1]
```diff
@@ -83,10 +92,61 @@ export async function createWorldsManagerComponent({
...
+    if (!permissionSetting.wallets.includes(address)) {
+      permissionSetting.wallets.push(address)
+    }
+    await storeWorldMetadata(worldName, { permissions: metadata!.permissions })
+  }
+
+  async function deleteAddressFromAllowList(worldName: string, permission: Permission, address: string): Promise<void> {
+    const metadata = await getMetadataForWorld(worldName)
+
+    const permissionSetting = metadata!.permissions[permission] as AllowListPermissionSetting
```
[the remark this answers is not in the document]
Comment (marianogoldman) [reply src/adapters/worlds-manager.ts:139]: Done

File: src/adapters/permissions-manager.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    const permissions = metadata?.permissions || defaultPermissions()
+    permissions[permission] = { type, ...extraOptions }
+    await worldsManager.storePermissions(worldName, permissions)
+  }
+
+  async function addAddressToAllowList(worldName: string, permission: Permission, address: string): Promise<void> {
+    const metadata = await worldsManager.getMetadataForWorld(worldName)
+
+    const permissionSetting = metadata!.permissions[permission] as AllowListPermissionSetting
```
Comment (hugoArregui) [inline src/adapters/permissions-manager.ts:35]: why can't metadata be undefined?

Comment (hugoArregui) [reply src/adapters/permissions-manager.ts:35]: I also don't think you need the `as ...` if you move it after the check

Comment (marianogoldman) [reply src/adapters/permissions-manager.ts]: fixed

**Review by hugoArregui — APPROVED** [review]

**Pull request merged**